pkg/app/pipedv1/notifier/slack.go

```diff
@@ -205,20 +205,21 @@ func (s *slack) buildSlackMessage(event model.NotificationEvent, webURL string)
 		fields            []slackField
 	)
 
-	generateDeploymentEventData := func(d *model.Deployment, accounts string) {
+	generateDeploymentEventData := func(d *model.Deployment, accounts string, groups string) {
```

Comment:
[IMO]
How about calling getAccountsAsString(), getGroupsAsString() here to simplify the switch part?
```suggestion
	generateDeploymentEventData := func(d *model.Deployment, accounts, groups []string) {
	accountsStr := getAccountsAsString(accounts)
	groupsStr := getGroupsAsString(groups)
```

Reply:
i dont get this 
- why we remove data type of `accounts`
- why add data type groups as list string?

Reply:
i believed we get return all method here as `string`